File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    
    for(int i = 0; i < numbers.size(); i++) {
        for(int j = i + 1; j < numbers.size(); j++) {
            int temp = numbers[i] + numbers[j];
            
            auto it = find(answer.begin(), answer.end(), temp);
            if(it != answer.end()) continue;
            answer.push_back(temp);
        }
    }

    sort(answer.begin(), answer.end());
    return answer;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] numbers) {
        List<int> answer = new List<int>();
        
        for(int i = 0; i < numbers.Length;i++) {
            for(int j = i + 1; j < numbers.Length; j++) {
                int temp = numbers[i] + numbers[j];
                if(answer.Contains(temp)) continue;
                answer.Add(temp);
            }
        }
        
        answer.Sort();
        
        return answer.ToArray();
    }
}